src/integrate/orchestrator.rs

```diff
@@ -0,0 +1,1204 @@
+            status: Status::Yellow,
+            name: "stacks-explorer".into(),
+            comment: "preparing container".into(),
+        }));
+        let _ = event_tx.send(DevnetEvent::info(format!("Starting stacks-explorer")));
+        match self.boot_stacks_explorer_container().await {
+            Ok(_) => {}
+            Err(message) => {
+                println!("{}", message);
+                self.terminate().await;
+                std::process::exit(1);
+            }
+        };
+        let _ = event_tx.send(DevnetEvent::ServiceStatus(ServiceStatusData {
+            order: 3,
+            status: Status::Green,
+            name: "stacks-explorer".into(),
+            comment: format!("http://0.0.0.0:{}", stacks_explorer_port),
+        }));
+
+        // Start bitcoin-explorer
+        let _ = event_tx.send(DevnetEvent::ServiceStatus(ServiceStatusData {
+            order: 4,
+            status: Status::Yellow,
+            name: "bitcoin-explorer".into(),
+            comment: "preparing container".into(),
+        }));
+        let _ = event_tx.send(DevnetEvent::info(format!("Starting bitcoin-explorer")));
+        match self.boot_bitcoin_explorer_container().await {
+            Ok(_) => {}
+            Err(message) => {
+                println!("{}", message);
+                self.terminate().await;
+                std::process::exit(1);
+            }
+        };
+        let _ = event_tx.send(DevnetEvent::ServiceStatus(ServiceStatusData {
+            order: 4,
+            status: Status::Green,
+            name: "bitcoin-explorer".into(),
+            comment: format!("http://0.0.0.0:{}", bitcoin_explorer_port),
+        }));
+
+        match terminator_rx.recv() {
+            Ok(true) => {
+                self.terminate().await;
+            }
+            Ok(false) => {
+                self.restart().await;
+            }
+            _ => {}
+        }
+    }
+
+    // if working_dir empty:
+    //      -> write config files
+    // else
+    //      -> read config files
+
+    pub async fn boot_bitcoin_node_container(&mut self) -> Result<(), String> {
+        let (docker, devnet_config) = match (&self.docker_client, &self.network_config) {
+            (Some(ref docker), Some(ref network_config)) => match network_config.devnet {
+                Some(ref devnet_config) => (docker, devnet_config),
+                _ => return Err("Unable to get devnet configuration".into()),
+            },
+            _ => return Err("Unable to get Docker client".into()),
+        };
+
+        let _info = docker
+            .create_image(
+                Some(CreateImageOptions {
+                    from_image: devnet_config.bitcoin_node_image_url.clone(),
+                    ..Default::default()
+                }),
+                None,
+                None,
+            )
+            .try_collect::<Vec<_>>()
+            .await
+            .map_err(|_| "Unable to create image".to_string())?;
```

Comment:
Might be useful to print out the container ID in debug or verbose logs, most advanced docker users would find this useful - [example](https://github.com/fussybeaver/bollard/blob/master/examples/image_from_scratch.rs#L75-L80)

Reply:
Addressed with latest